### PR TITLE
Spilt Ibft MessageValidator into components

### PR DESCRIPTION
**Pull request by rain-on**
Moving to IBFT2.1 requires that validation be conducted
on the signeddata aspects of a message separately from the
'piggybacked' block.

### CODE REVIEW

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+  private Optional<SignedData<ProposalPayload>> proposal = Optional.empty();
+
+  public SignedDataValidator(
+      final Collection<Address> validators,
+      final Address expectedProposer,
+      final ConsensusRoundIdentifier roundIdentifier,
+      final BlockValidator<IbftContext> blockValidator,
+      final ProtocolContext<IbftContext> protocolContext,
+      final BlockHeader parentHeader) {
```
Comment (jframe) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java]: doesn't look this is used

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/MessageValidator.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018 ConsenSys AG.
+ * Copyright 2019 ConsenSys AG.
```
Comment (CjHare) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/MessageValidator.java]: Usually copyright data is left as the earliest date i.e. prior art

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/MessageValidator.java [hunk 3]
```diff
@@ -12,205 +12,27 @@
...
-      return false;
-    }
-
-    if (!validators.contains(msg.getAuthor())) {
-      LOG.info(
-          "Invalid {} message, was not transmitted by a validator for the " + "associated round.",
-          msgType);
-      return false;
-    }
+  private final SignedDataValidator signedDataValidator;
```
Comment (jframe) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/MessageValidator.java]: Perhaps some tests for messageValidator to prove that it is delegating the validation of the payload to the signedDataValidator

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    }
+
+    if (!validateSignedProposalPayload(msg)) {
+      return false;
+    }
+
+    if (!validateBlockMatchesProposalRound(msg.getPayload())) {
+      return false;
+    }
+
```
Comment (CjHare) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java]:
```if(validateSignedProposalPayload(msg) && validateBlockMatchesProposalRound(msg.getPayload()){
  proposal = Optional.of(msg);
  return true;
}

return false;
```

Nicer to read? (removes negation, groups together logical conditions)

Comment (rain-on) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java]: Somewhat - but this follows the pattern which has been used throughout - i.e. final return is always true ...

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java (hunk 1 above)
Comment (rain-on) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/SignedDataValidator.java]: True, and this is part of a bigger issue - but will remove it from here (which is the start of something nice).

**Pull request merged**